### ISSUE TITLE
Where does 'samples_to_keep.txt' used as input in `04-mask_pseudogenome.sh` come from?

### ISSUE THREAD

**Issue by tavareshugo** [cambiotraining/bacterial-genomics#5]
After running the `bactmap` pipeline, I do not seem to have this file, which is used in `04-mask_pseudogenome.sh`: 

```bash
# text file of samples to keep
samplestokeep="results/bactmap/pseudogenomes_check/samples_to_keep.txt"
```

Are we supposed to generate this file ourselves? In which case, perhaps it should be elsewhere, not inside `bactmap`, as that implies it's an output from that pipeline.

**Comment by avantonder** [comment]
Sorry, that was included as some of the samples in the first TB dataset I used would be excluded. It's not necessary so I removed it from the script:

```
#!/bin/bash

#### Settings #####

# directory with pseudogenome FASTA

fasta_dir="results/bactmap/pseudogenomes"

# output directory for results
outdir="results/bactmap/masked_alignment"

# path to bed file with masking co-ordinates
bed="resources/masking/MTBC0_Goigetal_regions_toDiscard.bed"

#### End of settings ####

#### Analysis ####
# WARNING: be careful changing the code below

# exit upon any error
set -e

# create output directory
mkdir -p $outdir

# copy pseudogenome alignment to output directory
cp $fasta_dir/aligned_pseudogenomes.fas $outdir

# mask alignment with co-ordinates in bed file
remove_blocks_from_aln.py -a $outdir/aligned_pseudogenomes.fas -t $bed -o $outdir/aligned_pseudogenomes_masked.fas
```

**Comment by tavareshugo** [comment]
that works, thanks

**Comment by bsalehe** [comment]
This one didn't work for me when testing the scripts for M_tuberculosis as participant. Keep getting the following error, "scripts/04-mask_pseudogenome.sh: line 30: remove_blocks_from_aln.py: command not found" The python file does not exist in the scripts folder. This also makes iqtree to fail downstream as well as it needs the output from here. Was is it intentionally removed or just forgot to put this file for tuberculosis analysis in the "M_tuberculosis" folder? Thanks

**Comment by tavareshugo** [comment]
@bsalehe just checking did you activate `mamba activate remove_blocks`?

**Comment by avantonder** [comment]
@bsalehe @tavareshugo Was going to ask the same thing

**Comment by tavareshugo** [comment]
Perhaps we could add a comment to each script about which environment should be activated.

**Comment by avantonder** [comment]
Sounds like a good idea

**Comment by bsalehe** [comment]
> @bsalehe just checking did you activate `mamba activate remove_blocks`?

No I didn't do this, but i just activated it and got the same message "scripts/04-mask_pseudogenome.sh: line 30: scripts/remove_blocks_from_aln.py: No such file or directory"

**Comment by bsalehe** [comment]
> Perhaps we could add a comment to each script about which environment should be activated. (opened #9 for this)

I tried to add in the beginning of script 07-run_pairsnp.sh `mamba deactivate` and `mamba activate pairsnp` which also ideally should be in other scripts. However, when I do that I got error "Run 'mamba init' to be able to run mamba activate/deactivate
and start a new shell session. Or use conda to activate/deactivate.

Run 'mamba init' to be able to run mamba activate/deactivate
and start a new shell session. Or use conda to activate/deactivate.

scripts/07-run_pairsnp.sh: line 16: pairsnp: command not found". This does not happen when I run manually in the terminal the two mamba commands above. I don't understand why mamba is behaving in this way. Is it an instance setup issue or something else?

Cheers!
